File: solutions/php/hamming/1/hamming.php
<?php
/**
 * Distance calculates the Hamming distance between two DNA strands.
 *
 * @param string $a DNA strand (ex: GACTCCA)
 * @param string $b DNA strand (ex: CACTCCG)
 * @return int The hamming distance between the two strings
 */
function distance($a, $b)
{
    if (strlen($a) !== strlen($b)) {
        die('Invalid input');
    }

    $distance = 0;
    for ($i = 0; $i < strlen($a); $i++) {
        if ($a[$i] !== $b[$i]) {
            $distance++;
        }
    }

    return $distance;
}

File: solutions/go/hamming/1/hamming.go
// Package hamming calculates the Hamming Distance between two DNA strands
package hamming

import "errors"

// Distance returns the Hamming Distance of two DNA strands a, and b.
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("DNA strands aren't the same length")
	}

	dist := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			dist++
		}
	}
	return dist, nil
}
